packages/by-name/microsoft/genpolicy/0007-genpolicy-support-for-VOLUME-definition-in-container.patch

```diff
@@ -0,0 +1,631 @@
+         // containers:
+@@ -159,6 +148,15 @@ impl yaml::K8sResource for StatefulSet {
+                 );
+             }
+         }
++
++        yaml::get_container_mounts_and_storages(
++            policy_mounts,
++            storages,
++            persistent_volume_claims,
++            container,
++            settings,
++            &self.spec.template.spec.volumes,
++        );
+     }
+ 
+     fn generate_policy(&self, agent_policy: &policy::AgentPolicy) -> String {
+diff --git a/src/tools/genpolicy/src/yaml.rs b/src/tools/genpolicy/src/yaml.rs
+index c898240af337f3cb7cfc34fa1398cb5a6bd828a5..07ebb32aea0ae8265c8deb8c32fb02242d1a7d84 100644
+--- a/src/tools/genpolicy/src/yaml.rs
++++ b/src/tools/genpolicy/src/yaml.rs
+@@ -251,24 +251,41 @@ pub fn get_container_mounts_and_storages(
+     persistent_volume_claims: &[pvc::PersistentVolumeClaim],
+     container: &pod::Container,
+     settings: &settings::Settings,
+-    volumes: &Vec<volume::Volume>,
++    volumes_option: &Option<Vec<volume::Volume>>,
+ ) {
+-    if let Some(volume_mounts) = &container.volumeMounts {
+-        for volume in volumes {
+-            for volume_mount in volume_mounts {
+-                if volume_mount.name.eq(&volume.name) {
+-                    mount_and_storage::get_mount_and_storage(
+-                        settings,
+-                        policy_mounts,
+-                        storages,
+-                        persistent_volume_claims,
+-                        volume,
+-                        volume_mount,
+-                    );
++    if let Some(volumes) = volumes_option {
++        if let Some(volume_mounts) = &container.volumeMounts {
++            for volume in volumes {
++                for volume_mount in volume_mounts {
++                    if volume_mount.name.eq(&volume.name) {
++                        mount_and_storage::get_mount_and_storage(
++                            settings,
++                            policy_mounts,
++                            storages,
++                            persistent_volume_claims,
++                            volume,
++                            volume_mount,
++                        );
++                    }
+                 }
+             }
+         }
+     }
++
++    // Add storage and mount for each volume defined in the docker container image
++    // configuration layer.
++    if let Some(volumes) = &container.registry.config_layer.config.Volumes {
++        for volume in volumes {
++            debug!("get_container_mounts_and_storages: {:?}", &volume);
++
++            mount_and_storage::get_image_mount_and_storage(
++                settings,
++                policy_mounts,
++                storages,
++                volume.0,
++            );
++        }
++    }
+ }
+ 
+ /// Add the "io.katacontainers.config.agent.policy" annotation into
+diff --git a/tests/integration/kubernetes/k8s-policy-deployments.bats b/tests/integration/kubernetes/k8s-policy-deployments.bats
```

Comment:
Consider removing the test scripts from the patch - we're not going to use them.